src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/sampler/PostWriter.java

```diff
@@ -299,7 +296,7 @@ public void setHeaders(URLConnection connection, HTTPSamplerBase sampler) throws
 
                     // Just append all the parameter values, and use that as the post body
                     StringBuilder postBodyBuffer = new StringBuilder();
-                    for (JMeterProperty jMeterProperty : sampler.getArguments()) {
+                    for (JMeterProperty jMeterProperty : sampler.getArguments().getEnabledArguments()) {
```

Comment:
It looks like we should not filter "empty" arguments from here.

Frankly, I do not understand the reason why "empty named" (`.isSkippable(...)`) arguments are included only in certain cases.